test/ao/sparsity/test_data_sparsifier.py

```diff
@@ -235,3 +237,216 @@ def test_nn_embeddings(self):
+            test_sparsifier = DataNormSparsifier(sparsity_level=0.5, sparse_block_shape=(1, 4), zeros_per_block=4)
+        for _ in range(iters_before_collapse):
+            new_data = torch.randn(20, 20)
+            test_sparsifier.add_data(name='test_data', data=new_data)
+            test_sparsifier.step()
+            mask = test_sparsifier.get_mask(name='test_data')
+            mask = mask.to(torch.float)
+            assert (1.0 - mask.mean().item()) > 0  # some sparsity achieved
+
+    def test_step_2_of_4(self):
+        # overriding default config for test purposes
+        default_config = {'sparsity_level': 1.0, 'zeros_per_block': 2, 'sparse_block_shape': (1, 4)}
+        data_list = [('test_data', torch.randn(4, 4))]
+        if self.norm_type == 'L1':
+            sparsifier = DataNormSparsifier(data_list=data_list, **default_config)
+        sparsifier.step()
+
+        for some_data in data_list:
+            name, data = some_data
+            mask = sparsifier.get_mask(name=name)
+            mask = mask.to(torch.float)
+            self.assertAlmostEqual(1.0 - mask.mean().item(), 0.5, places=2)
+            for row in mask:
+                for idx in range(0, len(row), 4):
+                    block = row[idx:idx + 4]
+                    block, _ = block.sort()
+                    assert (block[:2] == 0).all()
+                    assert (block[2:] != 0).all()
+
+    def test_sparsity_level(self):
+        sparsity_levels = [-1.0, 0.0, 0.5, 1.0, 2.0]
+        sparse_block_shapes = [(1, 1), (1, 4), (2, 2), (4, 1)]
+        zeros_per_blocks = [0, 1, 2, 3, 4]
+        sparsifier = self.get_sparsifier()
+        testcases = itertools.tee(itertools.product(sparsity_levels,
+                                                    sparse_block_shapes,
+                                                    zeros_per_blocks))
+
+        # get some data
+        name, data, _ = self.data_with_config['name'], self.data_with_config['data']
+        for sl, sbs, zpb in testcases[0]:
+            new_name = f'{name}_{sl}_{sbs}_{zpb}'
+            current_config = {'sparsity_level': sl, 'sparse_block_shape': sbs, 'zeros_per_block': zpb}
+            sparsifier.add_data(name=new_name, data=data, **current_config)
+            if zpb > sbs[0] * sbs[1]:
+                continue
+
+        sparsifier.step()
+        sparsifier.squash_mask()
+        for sl, sbs, zpb in testcases[0]:
+            new_name = f'{name}_{sl}_{sbs}_{zpb}'
+            sparsified_data = sparsifier.get_data(name=new_name, original=False)
+            # sparse mask
+            sparse_mask = (sparsified_data == 0).float()
+            if zpb == 0:
+                assert sparse_mask.mean() == 0
+            else:
+                # Ratio of individual zeros in the tensor
+                true_sl = min(max(sl, 0.0), 1.0)
+                true_sl = true_sl * zpb / sbs[0] / sbs[1]
+                assert sparse_mask.mean() == true_sl
+
+    def run_tests(self):
```

Comment:
I feel this is structured differently from what we have in other code, for example in https://github.com/pytorch/pytorch/blob/master/test/quantization/fx/test_quantize_fx.py#L702 we just define a TestCase and then some methods in the TestCase class which implements a single test. can we align with that structure?

Reply:
Do you mean that we convert each function inside the runner class into a separate class and invoke each one of them in the main test case class?

Reply:
added a TODO comment, will change structure in a later PR